`no-undefined-types` cannot find a type created in JavaScript

## Expected behavior

The `@typedef` tag cannot find a type created in JavaScript. No error should be reported by [`no-undefined-types`](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/docs/rules/no-undefined-types.md).

## Actual behavior

This bug has been present since [v60.4.0](https://github.com/gajus/eslint-plugin-jsdoc/releases/tag/v60.4.0).

- `npm install`
- `npx eslint`
  ```
  /home/regseb/testcase/index.js
    14:1  error  The type 'Severities' is undefined  jsdoc/no-undefined-types
    14:1  error  The type 'Severities' is undefined  jsdoc/no-undefined-types

  ✖ 2 problems (2 errors, 0 warnings)
  ```

## `eslint.config.js`

```js
import jsdoc from "eslint-plugin-jsdoc";

export default [
  {
    plugins: { jsdoc },
    rules: {
      "jsdoc/no-undefined-types": "error",
    }
  }
];
```

## `index.js`

```js
/**
 * La liste des sévérités.
 *
 * @type {Object<string, number>}
 */
const Severities = {
    FATAL: 1,
    ERROR: 2,
    WARN: 3,
    INFO: 4,
};

/**
 * @typedef {Severities[keyof Severities]} Severity Le type des sévérités.
 */

export default Severities;
```

## `package.json`

```json
{
  "name": "testcase",
  "version": "1.0.0",
  "type": "module",
  "dependencies": {
    "eslint": "9.38.0",
    "eslint-plugin-jsdoc": "61.1.7"
  }
}
```

## Environment

- Node version: v22.20.0
- ESLint version: v9.38.0
- `eslint-plugin-jsdoc` version: 61.1.7

--- a/docs/rules/no-undefined-types.md
+++ b/docs/rules/no-undefined-types.md
@@ -1097,5 +1097,23 @@ class Storage {
   /** @type {globalThis.localStorage} */
   #storage
 }
+
+/**
+ * La liste des sévérités.
+ *
+ * @type {Object<string, number>}
+ */
+const Severities = {
+    FATAL: 1,
+    ERROR: 2,
+    WARN: 3,
+    INFO: 4,
+};
+
+/**
+ * @typedef {Severities[keyof Severities]} Severity Le type des sévérités.
+ */
+
+export default Severities;
 ````
 

--- a/src/rules/noUndefinedTypes.js
+++ b/src/rules/noUndefinedTypes.js
@@ -342,7 +342,10 @@ export default iterateJsdoc(({
                 break;
               }
 
-              return [];
+              // Module scope names are also defined
+              return [
+                name,
+              ];
           }
 
           return [

--- a/test/rules/assertions/noUndefinedTypes.js
+++ b/test/rules/assertions/noUndefinedTypes.js
@@ -1870,5 +1870,26 @@ export default /** @type {import('../index.js').TestCases} */ ({
         },
       },
     },
+    {
+      code: `
+        /**
+         * La liste des sévérités.
+         *
+         * @type {Object<string, number>}
+         */
+        const Severities = {
+            FATAL: 1,
+            ERROR: 2,
+            WARN: 3,
+            INFO: 4,
+        };
+
+        /**
+         * @typedef {Severities[keyof Severities]} Severity Le type des sévérités.
+         */
+
+        export default Severities;
+      `,
+    },
   ],
 });

